mempool/clist_mempool.go

```diff
@@ -548,6 +548,38 @@ func (mem *CListMempool) ReapMaxBytesMaxGas(maxBytes, maxGas int64) types.Txs {
 	return txs
 }
 
+// Safe for concurrent use by multiple goroutines.
+// Used in Microblocks implementation
+func (mem *CListMempool) ReapOneTx(maxBytes, maxGas int64) types.Txs {
```

Comment:
I feel like I have mentioned this somewhere else before but one of these Tx can potentially contain a lot of state transitions. I think it's OK to ignore this for this fact for this first PR but IMO we should additionally make use of that maxGas/maxBytes param to further limit the amount of state transitions in a microblock.